### DISCUSSION TITLE
peak2gene using geneScore

### DISCUSSION THREAD

**Discussion by Brawni** [GreenleafLab/ArchR#873] [Questions / Documentation] [answered]
Hello,

Im wondering whether running the peak2gene analysis correlating distal peaks to genescore instead of gene expression would be still interesting once the genescore has been recomputed to sum up only fragments in promoter and or overlapping gene body? Is there an easy way to do this?

Thank you!

**Comment by rcorces** [comment, marked as answer]
I've never done this and there are obvious reasons why this could be problematic but I think you should be able to do what you want just by setting `useMatrix` to `GeneScoreMatrix`.
Related, the parameter documention for `addPeak2GeneLinks()` was incomplete and I updated it via https://github.com/GreenleafLab/ArchR/commit/2aabfc349fcd403388cc26a5f3fd0821b33a9764

**Reply by Brawni** [reply]
I see thanks. But if you recompute the genescore to consider only fragments in promoter / gene body and then run the analysis would that still be problematic? and if yes why?

**Reply by rcorces** [reply]
I havent thought this through completely but I think there are certain precautions you would need to take. For example, it could be problematic if your regulatory element is an intronic enhancer, this will contribute to the gene score.

**Reply by Brawni** [reply]
So if I only wanted to use fragments mapping to the promoter region of the gene as a proxy for gene expression, what should it be the correct settings in the `addGeneScoreMatrix` ?

**Reply by rcorces** [reply]
`useTSS = TRUE` will disable gene body extension.
I think there is actually a bug in the code (or documentation) for this use case though.
You'll have to also set `extendTSS=TRUE` and this will extend the TSS region on either side based on `geneUpstream` and `geneDownstream` as shown here:
https://github.com/GreenleafLab/ArchR/blob/968e4421ce7187a8ac7ea1cf6077412126876d5f/R/MatrixGeneScores.R#L224
However, the documentation would make you believe that `extendUpstream` and `extendDownstream` would be the proper params for this.

**Reply by Brawni** [reply]
ok thanks a lot! I will give it a go!
